### PR TITLE
Scope auto-generated gem routes

### PR DESCRIPTION
**Pull request by bradhaydon**
In order to avoid any Rails route conflicts, the auto-generated Hangar routes should be be scoped. This is an alternate approach to pull request faradayio/hangar#13.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Hangar.do_not_delete = %w(very_important_things valuable_bitcoin_keys spatial_re
 
 ### Specifying DatabaseCleaner deletion strategy
 
-Not all ORMs support the deletion strategy, such as mongoid. You can override the clean strategy being used by the `DELETE /` request as follows:
+Not all ORMs support the deletion strategy, such as mongoid. You can override the clean strategy being used by the `DELETE /factori` request as follows:
```
Comment (rossmeissl) [inline README.md]: @bradhaydon is this a typo?

**Comment by rossmeissl** [comment]
Hi @bradhaydon as I said in #13 I don't really understand how there could be a route conflict if the Hangar routes are constrained by https://github.com/faradayio/hangar/blob/master/lib/hangar/constraint.rb ?

File: README.md (hunk 1 above)
Comment (bradhaydon) [reply README.md]: You're correct that is a typo; good catch! I have pushed up an updated README.md file.

**Comment by rossmeissl** [comment]
@bradhaydon got it, thanks. We prefer unscoped. Perhaps we could just add scoping as an option? For instance, if `ENV['HANGAR_SCOPE']`, use that, otherwise leave unscoped?

**Comment by bradhaydon** [comment]
I love that idea! I’ll create the change and push it up shortly.

**Comment by bradhaydon** [comment]
@rossmeissl I added a the following configuration to the gem: 

```ruby
# config/initializers/hangar.rb
Hangar.route_namespace = :factories
```

which will generate the following routes for the example **posts** factory:

```
GET /factories/posts
GET /factories/posts/new
DELETE /factories
```

I would love to hear your thoughts on this approach.

**Comment by rossmeissl** [comment]
Fabulous @bradhaydon thank you!

**Pull request merged**